packages/social-previews/src/facebook-preview/hooks/use-image-hook.ts

```diff
@@ -1,28 +1,30 @@
 import { __ } from '@wordpress/i18n';
 import { useCallback, useState } from 'react';
 import { LANDSCAPE_MODE, PORTRAIT_MODE } from '../../constants';
+import type { ImageMode } from '../../types';
 
-type Mode = typeof LANDSCAPE_MODE | typeof PORTRAIT_MODE | undefined;
 type ImageEventHandler = ( event: React.SyntheticEvent< HTMLImageElement > ) => void;
 type ImgProps = {
 	alt: string;
 	onLoad: ImageEventHandler;
 	onError: ImageEventHandler;
 };
-type UseImage = () => [ Mode, boolean, ImgProps ];
+type UseImage = ( mode?: ImageMode ) => [ ImageMode | undefined, boolean, ImgProps ];
```

Comment:
This is more of a thought than a suggestion, but do you think `useImage` should take an object rather than just the mode? That way the code could be a bit more explicit:

`const [ mode, isLoadingImage, imgProps ] = useImage( { mode: LANDSCAPE_MODE } );`

Reply:
Good call!